core/src/identity/secp256k1.rs

```diff
@@ -122,36 +118,38 @@ impl SecretKey {
     /// ECDSA signature, as defined in [RFC3278].
     ///
     /// [RFC3278]: https://tools.ietf.org/html/rfc3278#section-8.2
-    pub fn sign(&self, msg: &[u8]) -> Vec<u8> {
-        let m = Message::from_slice(Sha256::digest(&msg).as_ref())
-            .expect("digest output length doesn't match secp256k1 input length");
-        SECP.sign(&m, &self.0).serialize_der()
+    pub fn sign(&self, msg: &[u8]) -> Result<Vec<u8>, SigningError> {
+        let m = Message::parse_slice(Sha256::digest(msg).as_ref())
+            .map_err(|_| SigningError::new("failed to parse secp256k1 digest"))?;
```

Comment:
Why did you choose not to preserve the source of the error in this and a few other places involving parsing?

Reply:
Mostly because the `source` is required to satisfy `Error + Send + Sync + 'static` and those errors do not.